### PR TITLE
Client creation docs

### PR DESCRIPTION
**Pull request by djmitche**
The `.taskcluster.yml` will automatically upload this for incorporation into the docs site.

### CODE REVIEW

File: docs/creating-clients.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+title: Creating Clients
+---
+
+The process for [getting user
+credentials](https://docs.taskcluster.net/reference/integrations/taskcluster-login/docs/getting-user-creds)
+is designed for other frontend applications that can use OIDC flow.  That does
```
Comment (eliperelman) [inline docs/creating-clients.md]: Replace double spaces with single spaces.

File: docs/creating-clients.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+format defined by [taskcluster-client's
+fromNow](https://docs.taskcluster.net/reference/libraries/taskcluster-client#relative-date-time-utilities).
+For example, `?expires=3h` would result in credentials expiring in 3 hours.
+
+Finally, `callback_url` is the URL to which the user's browser should be
+redirected with the resulting client. It is called with the query parameters
+`clientId` and `accessToken`.  When this URL is accessed, the tool should
+respond with code to close the browser window, thereby hiding the access token:
+
+```
```
Comment (eliperelman) [inline docs/creating-clients.md]: Change to:

```
<!doctype html>
<html>
<body>
  <h1>Signed In</h1>
  <script>
    // Close the window after 500ms
    setTimeout(() => window.close(), 500);
  </script>
</body>
</html>
```

File: docs/creating-clients.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Tools provides an alternative flow to support non-OIDC applications. The
+process involves a browser and a redirect flow that finishes at a URL pointing
+to a local server, containing Taskcluster credentials. For command-line tools,
+this is a small webserver running within the tool itself.  The tool itself
+requests that the user either create a new client or reset the `accessToken`
+for an existing client.
+
+The tool should open a new browser window with a URL with prefix
+`https://tools.taskcluster.net/auth/clients/new` and the following query parameters:
```
Comment (eliperelman) [inline docs/creating-clients.md]: Add an adjective for "URL-encoded", i.e. "and the following URL-encoded query parameters:".

File: docs/creating-clients.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+the newly-created user.
+
+The `scope` parameter can be specified multiple times, and allows the creation
+of a client with only the required scopes -- a good security practice. The
+resulting client's scopes are the *intersection* of the user's scopes and the
+`scope` parameters. For example, a tool for administering the hooks service
+might request `?scope=hooks:*&scope=assume:hook-id:*`.
+
+The `expires` parameter specifies a lifetime for the resulting clientId, in a
+format defined by [taskcluster-client's
```
Comment (eliperelman) [inline docs/creating-clients.md]: Should we link to the project we are actually using, like https://github.com/taskcluster/taskcluster-client-web#relative-date-time-utilities?

File: docs/creating-clients.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+		setTimeout(function() {
+			window.close();
+		}, 500);
+	</script>
+</head>
+<body>
+	<h1>Signed In</h1>
+</body>
+</html>
+````
```
Comment (eliperelman) [inline docs/creating-clients.md]: There are 4 backticks here.

File: docs/creating-clients.md (hunk 2 above)
Comment (eliperelman) [inline docs/creating-clients.md]: Does this need the syntax highlighting language with `html`?

File: docs/creating-clients.md (hunk 4 above)
Comment (djmitche) [reply docs/creating-clients.md]: It should be the same.

File: docs/creating-clients.md (hunk 2 above)
Comment (djmitche) [reply docs/creating-clients.md]: In fact, closing the window doesn't work, so I should just delete that.

Comment (djmitche) [reply docs/creating-clients.md]: I don't think we get syntax highlighting

File: docs/creating-clients.md (hunk 3 above)
Comment (djmitche) [reply docs/creating-clients.md]: Please use a comma after `i.e.,`

**Comment by eliperelman** [comment]
This is a docs-only change, so no risk. Feel free to merge when convenient.

**Pull request merged**